console/src/workspace/ontology.tsx

```diff
@@ -6,118 +6,135 @@
-    const next = Tree.removeNode({
-      tree: nodes,
-      keys: resources.map((r) => r.id.toString()),
-    });
-    setNodes([...next]);
-  })();
-};
-
-const handleCreateNewSchematic = ({
-  client,
-  services,
-  placeLayout,
-  selection,
-  state: { nodes, setNodes, resources, setResources },
-}: Ontology.TreeContextMenuProps): void => {
-  const ws = selection.resources[0].id.key;
-  void (async () => {
-    const schematic = await client.workspaces.schematic.create(ws, {
-      name: "New Schematic",
-      snapshot: false,
-      data: deep.copy(Schematic.ZERO_STATE) as unknown as UnknownRecord,
-    });
-    const otg = await client.ontology.retrieve(
-      new ontology.ID({ key: schematic.key, type: "schematic" }),
-    );
-    placeLayout(
-      Schematic.create({
-        ...(schematic.data as unknown as Schematic.State),
-        key: schematic.key,
-        name: schematic.name,
-        snapshot: schematic.snapshot,
-      }),
-    );
-    setResources([...resources, otg]);
-    const nextNodes = Tree.setNode({
-      tree: nodes,
-      destination: selection.resources[0].key,
-      additions: Ontology.toTreeNodes(services, [otg]),
-    });
-    setNodes([...nextNodes]);
-  })();
-};
-
 const TreeContextMenu: Ontology.TreeContextMenu = (props): ReactElement => {
   const {
-    selection: { resources },
+    selection,
+    client,
+    services,
+    placeLayout,
+    store,
+    state: { nodes, setNodes, resources, setResources },
   } = props;
-  const clusterKey = Cluster.useSelectActiveKey();
 
-  const handleSelect = (key: string): void => {
-    switch (key) {
-      case "delete":
-        return handleDelete(props);
-      case "rename":
-        return Tree.startRenaming(resources[0].id.toString());
-      case "group":
-        void Group.fromSelection(props);
-        return;
-      case "plot":
-        // TODO: actually implement this case
-        return;
-      case "schematic": {
-        return handleCreateNewSchematic(props);
+  const handleDelete = (): void => {
+    void (async () => {
+      const s = store.getState();
+      const activeKey = selectActiveKey(s);
+      const active = selection.resources.find((r) => r.id.key === activeKey);
+      if (active != null) {
+        store.dispatch(setActive(null));
+        store.dispatch(Layout.clearWorkspace());
       }
-      case "link": {
-        const toCopy = `synnax://cluster/${clusterKey}/workspace/${resources[0].id.key}`;
-        void navigator.clipboard.writeText(toCopy);
-        return;
-      }
-    }
+      await client.workspaces.delete(...selection.resources.map((r) => r.id.key));
+      const next = Tree.removeNode({
+        tree: nodes,
+        keys: selection.resources.map((r) => r.id.toString()),
+      });
+      setNodes([...next]);
+    })();
+  };
+
+  const handleCreateNewSchematic = (): void => {
+    const workspace = selection.resources[0].id.key;
+    void (async () => {
```

Comment:
We should improve this void async pattern. We can probably leverage something like a custom wrapper around a react query mutation to make things much cleaner and more reliable. We'll talk tomorrow.